packages/zen-bull-netting/src/ZenBullNetting.sol

```diff
@@ -427,6 +484,143 @@ contract ZenBullNetting is Ownable, EIP712 {
+            _checkOrder(_params.orders[i]);
+            _useNonce(_params.orders[i].trader, _params.orders[i].nonce);
+
+            require(!_params.orders[i].isBuying, "ZBN19");
+            require(_params.orders[i].price <= _params.clearingPrice, "ZBN20");
+
+            if (_params.orders[i].quantity < toExchange) {
+                toExchange -= _params.orders[i].quantity;
+                IERC20(oSqth).transferFrom(
+                    _params.orders[i].trader, address(this), _params.orders[i].quantity
+                );
+            } else {
+                IERC20(oSqth).transferFrom(_params.orders[i].trader, address(this), toExchange);
+                break;
+            }
+        }
+
+        uint256 usdcToRepay = _params.withdrawsToProcess
+            * IEulerSimpleLens(eulerLens).getDTokenBalance(usdc, zenBull) / bullTotalSupply;
+
+        // WETH-USDC swap
+        _exactOutFlashSwap(
+            weth,
+            usdc,
+            _params.wethUsdcPoolFee,
+            usdcToRepay,
+            _params.maxWethForUsdc,
+            uint8(FLASH_SOURCE.WITHDRAW_AUCTION),
+            abi.encodePacked(_params.withdrawsToProcess, oSqthAmount, usdcToRepay)
+        );
+
+        // send WETH to market makers
+        IWETH(weth).deposit{value: address(this).balance - initialEthBalance}();
+        toExchange = oSqthAmount;
+        uint256 oSqthQuantity;
+        for (uint256 i = 0; i < _params.orders.length && toExchange > 0; i++) {
+            if (_params.orders[i].quantity < toExchange) {
+                oSqthQuantity = _params.orders[i].quantity;
+            } else {
+                oSqthQuantity = toExchange;
+            }
+
+            IERC20(weth).transfer(
+                _params.orders[i].trader, (oSqthQuantity * _params.clearingPrice) / 1e18
+            );
+            toExchange -= oSqthQuantity;
+
+            emit WithdrawAuction(
+                _params.orders[i].trader,
+                _params.orders[i].bidId,
+                oSqthQuantity,
+                _params.clearingPrice
+                );
+        }
+
+        // send ETH to withdrawers
+        uint256 ethToWithdrawers = IWETH(weth).balanceOf(address(this));
+        IWETH(weth).withdraw(ethToWithdrawers);
+
+        uint256 remainingWithdraws = _params.withdrawsToProcess;
+        uint256 j = withdrawsIndex;
+        uint256 ethAmount;
+
+        while (remainingWithdraws > 0) {
+            Receipt memory withdraw = withdraws[j];
+            if (withdraw.amount == 0) {
+                j++;
+                continue;
+            }
+            if (withdraw.amount <= remainingWithdraws) {
+                // full usage
+                remainingWithdraws -= withdraw.amount;
+                zenBullBalance[withdraw.sender] -= withdraw.amount;
+
+                // send proportional usdc
+                ethAmount = withdraw.amount * ethToWithdrawers / _params.withdrawsToProcess;
```

Comment:
I think this is ok as it will round down, but its possible that folks get less ETH due to precision, just noting

example, 2 wei * 100 wei / 300 wei rounds down to 0

its ok as long as it doesn't round up